src/protocol/metadata.rs

```diff
@@ -4,7 +4,64 @@ use error::Result;
 use codecs::{AsStrings, ToByte, FromByte};
 
 use super::{HeaderRequest, HeaderResponse};
-use super::{API_KEY_METADATA, API_VERSION};
+use super::{API_KEY_API_VERSIONS, API_KEY_METADATA, API_VERSION};
+
+#[derive(Debug)]
+pub struct ApiVersionsRequest<'a> {
+    pub header: HeaderRequest<'a>,
+}
+
+impl<'a> ApiVersionsRequest<'a> {
+    pub fn new(correlation_id: i32, client_id: &'a str) -> ApiVersionsRequest<'a> {
+        ApiVersionsRequest {
+            header: HeaderRequest::new(API_KEY_API_VERSIONS,
+                                       API_VERSION,
```

Comment:
Oh, I see; I misunderstood how the API versioning worked. I just realized that every request type has its own individual version, which starts at 0, which will work out here by happenstance. This kind of makes me think that we should refactor the request code in general.

@xitep I think this presents an interesting problem. I was thinking we should make a ticket to update all of our request/response structs with the additional fields from new API versions, but if we do that, how do we deserialize responses from older APIs that don't have those fields? We could use `Option`s, but that's a very clunky option, I think. We could have separate structs like `ProduceResponseV0`, `ProduceResponseV1`, etc., but that's also kind of clunky. Man, maintaining compatibility across all the versions is such a pain 😰